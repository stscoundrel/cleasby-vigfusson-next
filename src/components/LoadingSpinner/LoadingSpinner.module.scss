.spinner {
  width: 75px;
  margin-bottom: 40px;
}