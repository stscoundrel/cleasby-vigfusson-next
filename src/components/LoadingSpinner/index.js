import styles from './LoadingSpinner.module.scss'

export default function LoadingSpinner() {
  return (
    <img src="/loading.svg" alt="" role="presentation" className={styles.spinner} />
  )
}
