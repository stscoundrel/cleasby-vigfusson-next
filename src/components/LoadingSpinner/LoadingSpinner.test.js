import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import LoadingSpinner from './index'

describe('Loading spinner component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<LoadingSpinner />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<LoadingSpinner />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
