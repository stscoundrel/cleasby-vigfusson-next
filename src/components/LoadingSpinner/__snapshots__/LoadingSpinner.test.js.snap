// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading spinner component Matches snapshot 1`] = `
<img
  alt=""
  className="spinner"
  role="presentation"
  src="/loading.svg"
/>
`;
